components/embed_text/src/main.py

```diff
@@ -61,9 +63,13 @@ def get_embedding_model(
         msg = f"Unknown provider {model_provider}"
         raise ValueError(msg)
 
-    @retry()  # make sure to keep trying even when api call limit is reached
+    # make sure to keep trying even when api call limit is reached
     def get_embeddings_vectors(self, texts):
-        return self.embedding_model.embed_documents(texts.tolist())
+        @retry(tries=self.tries, logger=logger)
+        def return_embeddings():
+            return self.embedding_model.embed_documents(texts.tolist())
+
+        return return_embeddings()
```

Comment:
FYI, you can just use a decorator like a normal function.

```suggestion
        retry_wrapper = retry(tries=self.tries, logger=logger)
        return retry_wrapper(self.embedding_model.embed_documents(texts.tolist()))
```

Reply:
🤯